PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.Daemon/FunctionApp/Stages/Analysis/TimerTriggerCronExpressionAnalyzer.cs

```diff
@@ -78,46 +83,71 @@ protected override void Run(IAttribute element, ElementProblemAnalyzerData data,
             var mayBeTimeSpanSchedule = literal.Contains(":");
             if (mayBeTimeSpanSchedule)
             {
-                if (!IsValidTimeSpanSchedule(literal, out var errorMessage))
+                if (IsValidTimeSpanSchedule(literal, out var errorMessage, out var description))
+                {
+                    consumer.AddHighlighting(new TimerTriggerCronExpressionHint(description, expressionArgument, expressionArgument.GetDocumentEndOffset()));
+                }
+                else if (!string.IsNullOrEmpty(description))
                 {
                     consumer.AddHighlighting(new TimerTriggerCronExpressionError(expressionArgument, errorMessage));
                 }
             }
-            else if (!IsValidCrontabSchedule(literal, out var errorMessage))
+            else
             {
-                consumer.AddHighlighting(new TimerTriggerCronExpressionError(expressionArgument, errorMessage));
+                if (IsValidCrontabSchedule(literal, out var errorMessage, out var description))
+                {
+                    consumer.AddHighlighting(new TimerTriggerCronExpressionHint(description, expressionArgument, expressionArgument.GetDocumentEndOffset()));
+                }
+                else if (!string.IsNullOrEmpty(description))
+                {
+                    consumer.AddHighlighting(new TimerTriggerCronExpressionError(expressionArgument, errorMessage));
+                }
             }
         }
 
-        private bool IsValidCrontabSchedule(string literal, [CanBeNull] out string errorMessage)
+        private bool IsValidCrontabSchedule(
+            string literal,
+            [CanBeNull] out string errorMessage,
+            [CanBeNull] out string description)
         {
             try
             {
                 var normalizedLiteral = NormalizeCronSchedule(literal);
-                CrontabSchedule.Parse(normalizedLiteral, new CrontabSchedule.ParseOptions {IncludingSeconds = true});
+                CrontabSchedule.Parse(normalizedLiteral, new CrontabSchedule.ParseOptions { IncludingSeconds = true });
                 errorMessage = null;
+                try
+                {
+                    description = ExpressionDescriptor.GetDescription(normalizedLiteral);
+                }
+                catch (Exception)
+                {
+                    description = null;
```

Comment:
Rider showed me that there may be a null ref exception; so maybe better will be a `""` value.

Reply:
Fixed another potential NRE though. Good catch!